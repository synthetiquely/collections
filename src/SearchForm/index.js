import React, { Component } from 'react';
import glamorous from 'glamorous';

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      term: '',
      loading: false,
      error: null,
    };
  }

  render() {
    return (
      <Form>
        <Input
          type="text"
          id="search"
          name="search"
          placeholder="Поиск по коллекциям"
        />
        <Button type="submit">Найти</Button>
      </Form>
    );
  }
}

const Form = glamorous.form({
  display: 'flex',
  flexDirection: 'row',
});

const Input = glamorous.input({
  minWidth: '100%',
  padding: '10px',
  fontFamily: 'Segoe UI, Tahoma, Geneva, Verdana, sans-serif',
  fontSize: '16px',
  color: '#000',
  border: '1px solid #ccc',
  ':hover,:active,:focus': {
    borderColor: '#ffdb4d',
  },
  outlineColor: '#ffdb4d',
});

const Button = glamorous.button({
  fontFamily: 'Segoe UI, Tahoma, Geneva, Verdana, sans-serif',
  fontSize: '16px',
  padding: '10px',
  color: '#000',
  border: '1px solid #ccc',
  borderLeft: 'none',
  ':hover,:active,:focus': {
    borderLeft: '1px solid',
    borderColor: '#ffdb4d',
  },
  backgroundColor: '#fff',
});

export default SearchForm;
