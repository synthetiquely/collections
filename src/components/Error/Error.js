import glamorous from 'glamorous';

const Error = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
  margin: '5px',
  padding: '5px',
  alignItems: 'center',
  backgroundColor: '#f8d7da',
  border: '1px solid #f5c6cb',
  borderRadius: '5px',
  textAlign: 'center',
  color: '#721c24',
});

export default Error;
