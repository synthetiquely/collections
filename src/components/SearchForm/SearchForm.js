import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import Form from '../styled/Form';
import TextInput from '../styled/TextInput';
import Button from '../styled/Button';

@inject('search')
@observer
class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      term: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    this.setState({
      term: event.currentTarget.value,
    });
  }

  onSubmit(event) {
    event.preventDefault();

    if (!this.state.term) {
      return;
    }

    this.props.search.search(this.state.term);
  }

  render() {
    return (
      <Form onSubmit={this.onSubmit}>
        <TextInput
          type="text"
          id="search"
          name="search"
          placeholder="Поиск по коллекциям"
          value={this.state.term}
          onChange={this.onChange}
        />
        <Button loading={this.props.search.isLoading} type="submit">
          Найти
        </Button>
      </Form>
    );
  }
}
export default SearchForm;
