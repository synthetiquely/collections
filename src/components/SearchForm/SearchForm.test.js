import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Provider } from 'mobx-react';
import SearchForm from './SearchForm';
import Error from '../Error/Error';

describe('Swipe Component', () => {
  const search = {
    api: {},
    collections: {},
    searchTerm: '',
    nextPage: 1,
    isLoading: false,
    error: null,
    loadItems: jest.fn(),
    fetchPhotos: jest.fn(),
    searchPhotos: jest.fn(),
    setSearchTerm: jest.fn(),
    setPage: jest.fn(),
    setLoading: jest.fn(),
    setError: jest.fn(),
    recalculateLimit: jest.fn(),
  };

  it('should render and match its own snapshot', () => {
    const component = (
      <Provider search={search}>
        <SearchForm />
      </Provider>
    );
    const wrapper = mount(component);
    const tree = renderer.create(component).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render Error component, if there is an error in the store', () => {
    const store = {
      error: 'Error',
    };
    const component = (
      <Provider search={store}>
        <SearchForm />
      </Provider>
    );
    const wrapper = mount(component);

    expect(wrapper.find(Error)).toHaveLength(1);
    expect(wrapper
      .find(Error)
      .find('h1')
      .text()).toEqual('Ошибка');
  });
});
