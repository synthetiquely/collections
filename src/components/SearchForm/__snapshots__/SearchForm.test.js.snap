// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swipe Component should render and match its own snapshot 1`] = `
<form
  className="css-9v06sd"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="css-3nor7q"
    id="search"
    name="search"
    onChange={[Function]}
    placeholder="Поиск по коллекциям"
    type="text"
    value=""
  />
  <button
    className="css-g7cb4b"
    type="submit"
  >
     
    <span
      style={
        Object {
          "padding": "0 15px",
        }
      }
    >
      Найти
    </span>
  </button>
</form>
`;
